dotnet-authserver/tests/TeacherIdentity.AuthServer.EndToEndTests/TrnTokenSignInJourney.cs

```diff
@@ -110,4 +114,524 @@ public async Task NewTeacherUser_WithTrnTokenChangesEmail_CreatesUserAndComplete
+
+        var trnTokenModel = await _hostFixture.TestData.GetTrnToken(trnToken.TrnToken);
+        Assert.Null(trnTokenModel?.UserId);
+    }
+
+    [Fact]
+    public async Task ExistingUserSignsInNoTrn_ValidTrnToken_AssignsToken()
+    {
+        await using var context = await _hostFixture.CreateBrowserContext();
+        var page = await context.NewPageAsync();
+
+        var user = await _hostFixture.TestData.CreateUser(userType: UserType.Default);
+        var differentEmail = _hostFixture.TestData.GenerateUniqueEmail();
+
+        var trnToken = await CreateValidTrnToken(email: differentEmail);
+
+        await page.StartOAuthJourney(CustomScopes.DqtRead, trnToken: trnToken.TrnToken);
+
+        await page.SignInFromTrnTokenLandingPage();
+
+        await page.SubmitEmailPage(user.EmailAddress);
+
+        await page.SubmitEmailConfirmationPage();
+
+        await page.SubmitCompletePageForExistingUser();
+
+        await page.AssertSignedInOnTestClient(user.EmailAddress, trnToken.Trn, user.FirstName, user.LastName);
+
+        var trnTokenModel = await _hostFixture.TestData.GetTrnToken(trnToken.TrnToken);
+        Assert.Equal(user.UserId, trnTokenModel?.UserId);
+
+        _hostFixture.EventObserver.AssertEventsSaved(
+            e =>
+            {
+                var userUpdatedEvent = Assert.IsType<UserUpdatedEvent>(e);
+                Assert.Equal(UserUpdatedEventChanges.Trn | UserUpdatedEventChanges.TrnLookupStatus, userUpdatedEvent.Changes);
+                Assert.Equal(trnTokenModel?.Trn, userUpdatedEvent.User.Trn);
+                Assert.Equal(user.UserId, userUpdatedEvent.User.UserId);
+            },
+            e => Assert.IsType<UserSignedInEvent>(e));
+    }
+
+    [Fact]
+    public async Task ExistingUserSignsIn_ValidTrnTokenNotMatchingTrn_SignInAndIgnoreToken()
+    {
+        await using var context = await _hostFixture.CreateBrowserContext();
+        var page = await context.NewPageAsync();
+
+        var user = await _hostFixture.TestData.CreateUser(userType: UserType.Default, hasTrn: true);
+
+        var differentTrn = _hostFixture.TestData.GenerateTrn();
+        var differentEmail = _hostFixture.TestData.GenerateUniqueEmail();
+
+        var trnToken = await CreateValidTrnToken(trn: differentTrn, email: differentEmail);
+
+        await page.StartOAuthJourney(CustomScopes.DqtRead, trnToken: trnToken.TrnToken);
+
+        await page.SignInFromTrnTokenLandingPage();
+
+        await page.SubmitEmailPage(user.EmailAddress);
+
+        await page.SubmitEmailConfirmationPage();
+
+        await page.SubmitCompletePageForExistingUser();
+
+        await page.AssertSignedInOnTestClient(user.EmailAddress, user.Trn, user.FirstName, user.LastName);
+
+        var trnTokenModel = await _hostFixture.TestData.GetTrnToken(trnToken.TrnToken);
+        Assert.Null(trnTokenModel?.UserId);
+    }
+
+    [Fact]
+    public async Task ExistingUserSignsIn_ValidTrnTokenMatchingTrn_SignInAndInvalidatesToken()
```

Comment:
This Invalidates the token rather than Ignores it. Follows the pattern of invalidating tokens when users sign in with same Trn